warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/edge/ProtobufEdgeDataTypeHandler.java

```diff
@@ -637,6 +635,9 @@ public long process(KEYIN key, RawRecordContainer event, Multimap<String,Normali
                     for (String subGroup : commonKeys) {
                         for (NormalizedContentInterface ifaceSource : mSource.get(subGroup)) {
                             for (NormalizedContentInterface ifaceSink : mSink.get(subGroup)) {
+                                if (ifaceSource.equals(ifaceSink)) {
```

Comment:
How can this happen?  Would this only happen if we have an edge with the same source and sink?  Would this happen if the value of the field is the same on both sides (in which case this may be ok)?  This may be ok I am just trying to understand why.

Reply:
We simply take the product of the source and sink fields.  In the case where we are trying to make edges between multiple instances of the same field (see the updated test tvmaze-edge-definition for an example), we were previously including the main diagonal if the source field name and group matched in the edge definition.  

This would only happen if all elements of the NCI are identical, not just if the values are equal.  We'd still make edges if the group or subgroup are different.

This comment made me realize I was still testing some of that. Thanks!

Now that I think about it, though, we should probably fix some of the processors before implementing this since they are not always subgroup smart...  But until then we will still produce duplicates edges to the field itself.